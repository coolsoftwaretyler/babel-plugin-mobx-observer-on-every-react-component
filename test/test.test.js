import { expect, test, describe } from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import { plugin } from '../src/index.js'


describe("Mobx Observer Babel Plugin", () => {
  test("single simple component without import", () => {
    const simpleReactComponent = `const Simple = () => <div>check</div>;`;
    const out = transform(simpleReactComponent, {
      plugins: [pluginSyntaxJsx, pluginTransReactJsx, plugin]
    });

    expect(out.code).toMatchSnapshot();
  })

  test("single simple component with import", () => {
    const simpleReactComponent = `import { observer } from "mobx-react";
    const Simple = observer(() => <div>check</div>);`;
    const out = transform(simpleReactComponent, {
      plugins: [pluginSyntaxJsx, pluginTransReactJsx, plugin]
    });

    expect(out.code).toMatchSnapshot();
  })
});