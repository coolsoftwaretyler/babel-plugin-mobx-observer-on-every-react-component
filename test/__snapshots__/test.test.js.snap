// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin single simple component without import 1`] = `
"import { observer } from "mobx-react";
const Simple = observer(() => <div>check</div>);"
`;

exports[`Mobx Observer Babel Plugin single simple component with import 1`] = `
"import { observer } from "mobx-react";
const Simple = observer(() => /*#__PURE__*/React.createElement("div", null, "check"));"
`;
